### PR TITLE
Add classified error counters to CloudWatch Logs sink

### PR DESCRIPTION
**Pull request by kiran536**
## Summary

- Add 3 classified error counters (`cloudWatchLogsAccessDenied`, `cloudWatchLogsResourceNotFound`, `cloudWatchLogsThrottled`) to the CloudWatch Logs sink, letting operators alarm on specific failure modes instead of relying on the single opaque `cloudWatchLogsRequestsFailed` total
- Place classification logic in a new `classifyAndCountFailure()` method in the `Uploader` inner class, with a defensive null-check on `awsErrorDetails` before reading `errorCode`
- Count `ResourceNotFoundException` at the terminal catch site rather than in `classifyAndCountFailure()` because the error is only terminal when auto-create is disabled or has already been attempted
- Keep all 3 counters additive to the existing `cloudWatchLogsRequestsFailed` counter so existing alarms remain unaffected

## Test plan

- [ ] Verify 3 new unit tests in `CloudWatchLogsMetricsTest` pass (counter initialization and increment)
- [ ] Verify 6 new unit tests in `CloudWatchLogsDispatcherTest` pass (access denied, throttling, terminal ResourceNotFound, successful auto-create does not increment ResourceNotFound, SdkClientException does not increment classified counters, null error details does not throw)
- [ ] Verify 4 new WireMock-based integration tests in `CloudWatchLogsMetricsIT` pass (counters materialize in a Micrometer registry without real AWS credentials)
- [ ] Run full `./gradlew :data-prepper-plugins:cloudwatch-logs:test` and confirm no regressions
- [ ] Confirm the optional real-AWS gated IT is skipped in CI and passes when credentials are available locally

### CODE REVIEW

File: data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java [hunk 1]
```diff
@@ -168,6 +169,7 @@ public void upload() {
                             // else branch and normal retry/DLQ logic takes over.
                         } else {
                             failureMessage = e.getMessage();
+                            cloudWatchLogsMetrics.increaseResourceNotFoundCounter(1);
```
Comment (Himtanaya) [inline data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java]: When `createLogStream`/`createLogGroup` is enabled and the customer is missing the `logs:CreateLogStream` (or `logs:CreateLogGroup`) permission, the `AccessDeniedException` is caught and swallowed inside `createResources()`:

```
} catch (CloudWatchLogsException | SdkClientException e) {
    LOG.warn("Unable to create log stream '{}/{}': {}", logGroupName, logStreamName, e.getMessage());
}
```

It doesn't increment any counter and the subsequent `PutLogEvents` retries then hit `ResourceNotFoundException`, which lands in the new else branch and increments `cloudWatchLogsResourceNotFound` + `cloudWatchLogsRequestsFailed`.

A permission problem (missing CreateLogStream) is reported as `ResourceNotFound`, while the actual `AccessDenied` root cause is invisible.

File: data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java [hunk 2]
```diff
@@ -229,6 +232,28 @@ private int handlePutLogEventsFailure(final Exception e, final int currentFailCo
...
+         */
+        private String classifyAndCountFailure(final Exception e) {
+            if (e instanceof AwsServiceException) {
+                final AwsServiceException ase = (AwsServiceException) e;
+                if (ase.isThrottlingException()) {
+                    cloudWatchLogsMetrics.increaseThrottledCounter(1);
+                    return "throttled";
+                }
+                if (ase.awsErrorDetails() != null
+                        && "AccessDeniedException".equals(ase.awsErrorDetails().errorCode())) {
```
Comment (Himtanaya) [inline data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java]: AWS isn't perfectly consistent about this string you'll see `AccessDenied` (no Exception suffix) from some paths, and CloudWatch Logs doesn't model access denied as a typed exception subclass (it surfaces as a base `CloudWatchLogsException` carrying the code in `awsErrorDetails()`), so there's no `instanceof` to fall back on. If the returned code ever differs from this one literal, `AccessDenied` silently degrades to unclassified and the new counter goes quiet with nothing failing to signal the regression.

We should match on `contains` instead of `equals`.

Comment (kiran536) [reply data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java]: You're right that AWS returns both `AccessDenied` and `AccessDeniedException` depending on the code path. I extracted an `isAccessDenied(AwsServiceException)` helper that uses `contains("AccessDenied")` instead of `equals("AccessDeniedException")`, which catches both variants. I added a unit test that verifies the `"AccessDenied"` (no suffix) error code on PutLogEvents is correctly classified and counted.

File: data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java (hunk 1 above)
Comment (kiran536) [reply data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java]: Good catch. I added a `classifyAndCountResourceCreationFailure()` helper that is called from both catch blocks in `createResources()`. When the exception is access-denied, it now increments the `cloudWatchLogsAccessDenied` metric and logs an "Access denied creating..." message, so permission problems during resource creation are no longer invisible. I also added unit tests for access-denied on both createLogStream and createLogGroup, and added a negative assertion to the existing `create_stream_throws_cloudwatch_logs_exception` test confirming that non-access-denied exceptions during creation do not increment the access-denied counter.

Comment (Himtanaya) [inline data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java]: `resourceNotFound` should only count when the sink isn't responsible for creating the resource. 
`createLogGroup`/`createLogStream` flags decide who owns provisioning. When they're off, the user owns the destination and an RNF is genuinely actionable so we should increase this counter. When they're on, the user delegated creation to the sink, so an RNF on `PutLogEvents` is just internal control flow before auto-create; the only meaningful outcomes are create-succeeds (no error), create-denied (→ accessDenied, which this PR now counts), or create-fails (→ requestsFailed). "Resource not found" is never the user's action item when they asked the sink to create it.

Maybe we should do something like:
```
} else {
    failureMessage = e.getMessage();
    if (!createLogGroup && !createLogStream) {
        cloudWatchLogsMetrics.increaseResourceNotFoundCounter(1);
    }
    failCount = handlePutLogEventsFailure(e, failCount, backoff);
}
```

File: data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java [hunk 3]
```diff
@@ -229,6 +232,34 @@ private int handlePutLogEventsFailure(final Exception e, final int currentFailCo
...
+         * Classifies an exception as access-denied, throttled, or unclassified and increments
+         * the corresponding classified counter. ResourceNotFound is NOT classified here — it is
+         * counted structurally at the terminal catch site. Returns the classification label for
+         * logging. Never throws.
+         */
+        private String classifyAndCountFailure(final Exception e) {
+            if (e instanceof AwsServiceException) {
+                final AwsServiceException ase = (AwsServiceException) e;
+                if (ase.isThrottlingException()) {
+                    cloudWatchLogsMetrics.increaseThrottledCounter(1);
```
Comment (Himtanaya) [inline data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java:247]: Can we also update the README with the new counters

File: data-prepper-plugins/cloudwatch-logs/build.gradle [hunk 4]
```diff
@@ -63,6 +65,7 @@ task integrationTest(type: Test) {
     testClassesDirs = sourceSets.integrationTest.output.classesDirs
 
     useJUnitPlatform()
+    forkEvery = 1
```
Comment (dlvenable) [inline data-prepper-plugins/cloudwatch-logs/build.gradle:68]: What is this for?

File: data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java (hunk 3 above)
Comment (kiran536) [reply data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/client/CloudWatchLogsDispatcher.java:247]: Done. Added all three new counters (cloudWatchLogsThrottled, cloudWatchLogsAccessDenied, cloudWatchLogsResourceNotFound) to the Metrics section of the README with descriptions of when each is incremented and its semantic meaning.

File: data-prepper-plugins/cloudwatch-logs/build.gradle (hunk 4 above)
Comment (kiran536) [reply data-prepper-plugins/cloudwatch-logs/build.gradle:68]: This isolates each integration test class in its own JVM fork to prevent Mockito stubbing state from leaking between test classes. Without it, the existing `CloudWatchLogsIT` setUp contamination causes `UnfinishedStubbingException` failures in our new `CloudWatchLogsMetricsIT` when both run in the same Gradle test worker.

**Review by dlvenable — APPROVED** [review]

**Review by bagmarnikhil — APPROVED** [review]
Thanks for the contribution. LGTM!

**Review by Himtanaya — APPROVED** [review]
